evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractCallOperation.java

```diff
@@ -191,29 +198,24 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
 
     final Account contract = frame.getWorldUpdater().get(to);
 
-    if (contract != null && contract.hasDelegatedCode()) {
-      if (contract.getCodeDelegationTargetCode().isEmpty()) {
-        throw new RuntimeException("A delegated code account must have delegated code");
-      }
-
-      if (contract.getCodeDelegationTargetHash().isEmpty()) {
-        throw new RuntimeException("A delegated code account must have a delegated code hash");
-      }
-
-      final long codeDelegationResolutionGas =
-          CodeDelegationGasCostHelper.codeDelegationGasCost(frame, gasCalculator(), contract);
-
-      if (frame.getRemainingGas() < codeDelegationResolutionGas) {
-        return new Operation.OperationResult(
-            codeDelegationResolutionGas, ExceptionalHaltReason.INSUFFICIENT_GAS);
-      }
+    final Account account = frame.getWorldUpdater().get(frame.getRecipientAddress());
```

Comment:
why reading this account before deductGasForCodeDelegationResolution ? 
can we check that after in order to avoid reading something useless in case of an InsufficientGasException ?